File: Back/TVShowsAPI.Models/ApiResponse.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace TVShowsAPI.Models
{
    public class ApiResponse<T>
    {
        public T Data { get; set; }
        public int Page { get; set; }
        public int Rows { get; set; }
        public int Counts { get; set; }
        public int Status { get; set; }
        public string ErrorMessage { get; set; }
    }
}

File: Front/TvShowsAngular/src/app/models/api-response.ts
export interface ApiResponse<T> {
  data: T;              // Datos genéricos de la respuesta
  page: number;         // Página actual de la paginación
  rows: number;         // Número de filas devueltas por página
  counts: number;       // Total de registros
  status: number;       // Código de estado HTTP de la respuesta
  errorMessage?: string; // Mensaje de error si ocurre alguno
}
